vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/Spreadsheet.java

```diff
@@ -4576,6 +4569,10 @@ void loadRangeComponents(HashSet<Component> newCustomComponents,
         setComponentIDtoCellKeysMap(_componentIDtoCellKeysMap);
     }
 
+    private String getComponentNodeId(Component component) {
+        return Integer.toString(component.getElement().getNode().getId());
```

Comment:
The node id needs to be updated (and old node ids cleaned up) whenever the spreadsheet gets reattached. This will ensure `@PreserveOnRefresh` works correctly.

Reply:
Oh. Looks like the issue is still there actually. I just needed to create the sheet + apply the custom editor fixture twice before trying.

Reply:
The fix and the relevant tests are included in the latest commits